app/src/main/java/nmd/nomad/PlacesListActivity.java

```diff
@@ -0,0 +1,111 @@
+package nmd.nomad;
+
+import android.content.Context;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.support.v7.app.ActionBarActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.ListView;
+import android.widget.ProgressBar;
+import android.widget.RelativeLayout;
+
+import java.util.ArrayList;
+
+import nmd.nomad.adapters.PlacesListArrayAdapter;
+import nmd.nomad.api.GomadClient;
+import nmd.nomad.api.ServiceGenerator;
+import nmd.nomad.models.Place;
+import retrofit.Callback;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+public class PlacesListActivity extends ActionBarActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_places_list);
+
+        // Set the listView and the adapter
+        ProgressBar progressBar = (ProgressBar) findViewById(R.id.progressBar);
+        ListView listView = (ListView) findViewById(R.id.placesListView);
+        LinearLayout noPlaceFound = (LinearLayout) findViewById(R.id.noPlaceFound);
+        final ArrayList<Place> placeList = new ArrayList<Place>();
+        final PlacesListArrayAdapter arrayAdapter = new PlacesListArrayAdapter(this, placeList);
+        listView.setAdapter(arrayAdapter);
+
+        // Set a callback to fill the listView
+        final Callback callback = new Callback() {
+            @Override
+            public void success(Object object, Response response) {
+                ArrayList<Place> places = (ArrayList<Place>) object;
+                progressBar.setVisibility(View.GONE);
+
+                if(places.isEmpty()) {
+                    listView.setVisibility(View.GONE);
+                    noPlaceFound.setVisibility(View.VISIBLE);
+                } else {
+                    noPlaceFound.setVisibility(View.GONE);
+                    listView.setVisibility(View.VISIBLE);
+                    arrayAdapter.clear();
+                    arrayAdapter.addAll(places);
+                    for (Place place : places) {
+                        System.out.println(place.toString());
+                    }
+                }
+            }
+
+            @Override
+            public void failure(RetrofitError retrofitError) {
+                System.out.println(retrofitError.getBody());
+            }
+        };
+
+        LocationManager locationManager = (LocationManager) this.getSystemService(Context.LOCATION_SERVICE);
+
+        // Define a listener that responds to location updates
+        LocationListener locationListener = new LocationListener() {
+            public void onLocationChanged(Location location) {
+                System.out.println(location.getLatitude() + ", " + location.getLongitude());
```

Comment:
Got it! Take a look at my new commit plz.

On Sat, Feb 7, 2015 at 2:12 PM, Michael Basil <notifications@github.com>
wrote:

> In app/src/main/java/nmd/nomad/PlacesListActivity.java
> <https://github.com/asm-products/nomad-android/pull/12#discussion_r24290348>
> :
>
> > +                    }
> > +                }
> > +            }
> > +
> > +            @Override
> > +            public void failure(RetrofitError retrofitError) {
> > +                System.out.println(retrofitError.getBody());
> > +            }
> > +        };
> > +
> > +        LocationManager locationManager = (LocationManager) this.getSystemService(Context.LOCATION_SERVICE);
> > +
> > +        // Define a listener that responds to location updates
> > +        LocationListener locationListener = new LocationListener() {
> > +            public void onLocationChanged(Location location) {
> > +                System.out.println(location.getLatitude() + ", " + location.getLongitude());
>
> I would recommend switching to using the Location provider from Google
> Play Services. You are currently using the one that is built into Android
> itself which isn't so accurate or quick.
>
> You can see how I am getting the last known location here:
> https://github.com/asm-products/nomad-android/blob/master/app/src/main/java/nmd/nomad/GooglePlacesApiActivity.java#L86
>
> (Although that is slightly advanced in that I am using RxJava and lambdas
> but you can probably just copy and paste the code.)
>
> Eventually this should search for a new location if there is no known last
> location but that would do for now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/asm-products/nomad-android/pull/12/files#r24290348>.
>